editor.planx.uk/src/@planx/components/MapAndLabel/Public/Context.tsx

```diff
@@ -152,13 +152,14 @@ export const MapAndLabelProvider: React.FC<MapAndLabelProviderProps> = (
 
   const addFeatureToMap = (geojson: GeoJSONChange) => {
     resetErrors();
-    setFeatures(geojson["EPSG:3857"].features);
-    setActiveIndex((features && features?.length - 2) || activeIndex + 1);
+    const newFeatures = geojson["EPSG:3857"].features;
+    setFeatures(newFeatures);
+    setActiveIndex(newFeatures.length - 1);
   };
 
   const addInitialFeaturesToMap = (features: Feature[]) => {
     setFeatures(features);
-    // TODO: setActiveIndex ?
+    // setActiveIndex(features.length - 1);
```

Comment:
This _looks_ like the right solution to me, and it does indeed mean the latest tab will be active/open when coming "back", but it unexpectedly make all other tabs un-clickable - so noted in PR description as outstanding bug